configman/config_manager.py

```diff
@@ -461,7 +461,10 @@ def write_conf(self, config_file_type, opener, skip_keys=None):
                 an_option = option_defs[a_key]
                 if ((not a_key.startswith('admin'))
                     and isinstance(an_option, Option)
-                    and an_option.secret
+                    and (
+                        an_option.secret
+                        or 'password' in an_option.name.lower()
+                    )
```

Comment:
What if you really don't want it to be a secret? E.g.
```python
a.add_option(
    'password_hint', 
    default="Your mother's maiden name?"
)
```

Generally I like this whole idea (of this PR) because it's better to be safe than explicit. However, it's now impossible to switch it off. Maybe it could work with something like
```python
... and (
    an_option.secret
    or (an_option.secret is None and 'password' in an_option.name.lower())
)
```